server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -829,6 +832,8 @@ public void onFailure(Exception e) {
                             }
                         }, updatedSnapshot.getRepositoryStateId(), false);
                     }
+                    assert updatedSnapshot.shards().size() == snapshot.shards().size()
```

Comment:
Can we assert in the constructor of SnapshotsInProgress.Entry that for every entry in "indices" there is at least one entry in the shards map and vice versa?

Reply:
Yea that sounds good, maybe do it in a follow-up since it requires a bit of a rewrite of `org.elasticsearch.snapshots.SnapshotsInProgressSerializationTests#randomSnapshot` (it creates all kinds of bogus instances wehre the indices don't match up) and it might be nice to have this change isolated?

Reply:
ok

Reply:
https://github.com/elastic/elasticsearch/pull/47598